src/chainstate/burn/distribution.rs

```diff
@@ -38,195 +39,301 @@ use util::log;
-    /// will be incorporated.  All other commits will be dropped.
+    /// If a burn refers to more than one commitment, its burn amount is *split* between those
+    ///   commitments
     ///
     /// Returns the distribution, which consumes the given lists of operations.
-    pub fn make_distribution(
-        all_block_candidates: Vec<LeaderBlockCommitOp>,
-        consumed_leader_keys: Vec<LeaderKeyRegisterOp>,
-        user_burns: Vec<UserBurnSupportOp>,
+    ///
+    /// * `block_commits`: this is a mapping from relative block_height to the block
+    ///     commits that occurred at that height. These relative block heights start
+    ///     at 0 and increment towards the present. When the mining window is 6, the
+    ///     "current" sortition's block commits would be in index 5.
+    /// * `block_commits`: this is a mapping from relative block_height to the user
+    ///     burns that occurred at that height. When the mining window is 6, the
+    ///     "current" sortition's user burns would be in index 5.
+    /// * `sunset_finished_at`: if set, this indicates that the PoX sunset finished before or
+    ///     during the mining window. This value is the first index in the block_commits
+    ///     for which PoX is fully disabled (i.e., the block commit has a single burn output).
+    pub fn make_min_median_distribution(
+        mut block_commits: Vec<Vec<LeaderBlockCommitOp>>,
+        user_burns: Vec<Vec<UserBurnSupportOp>>,
+        sunset_finished_at: Option<u8>,
     ) -> Vec<BurnSamplePoint> {
-        // trivial case
-        if all_block_candidates.len() == 0 {
-            return vec![];
-        }
-
-        BurnSamplePoint::ops_sanity_checks(&all_block_candidates, &user_burns);
-
-        // map each leader key's position in the blockchain to its index in consumed_leader_keys.
-        // note that this effectively de-duplicates entries in consumed_leader_keys (see below)
-        let mut key_index: BTreeMap<(u64, u32), usize> = BTreeMap::new();
-        for i in 0..consumed_leader_keys.len() {
-            let key_loc = (
-                consumed_leader_keys[i].block_height,
-                consumed_leader_keys[i].vtxindex,
-            );
-            assert!(
-                key_index.get(&key_loc).is_none(),
-                format!("duplicate entry: {:?}", &key_loc)
-            );
-
-            key_index.insert(key_loc, i);
-        }
-
-        // sanity check -- each block commit must refer to a consumed key
-        for i in 0..all_block_candidates.len() {
-            let bc = &all_block_candidates[i];
-            match key_index.get(&(bc.key_block_ptr as u64, bc.key_vtxindex as u32)) {
-                None => {
-                    panic!(
-                        "No leader key for block commitment {} at ({},{}) -- points to ({},{})",
-                        &bc.txid, bc.block_height, bc.vtxindex, bc.key_block_ptr, bc.key_vtxindex
+        // sanity check
```

Comment:
For clarity's sake, it might be worth putting the sanity check into its own private method.

Reply:
Updated in 5c80770d4